src/types.ts

```diff
@@ -35,21 +35,18 @@ type CreditCardTypeSecurityCodeLabel =
   | "CVP2";
 
 export type CreditCardType = {
-  niceType?: string;
-  type?: string;
-  patterns?: number[] | [number[]];
-  gaps?: number[];
-  lengths?: number[];
-  code?: {
-    size?: number;
-    name?: string;
+  niceType: string;
+  type: string;
+  patterns: number[] | [number[]];
+  gaps: number[];
+  lengths: number[];
+  code: {
+    size: number;
+    name: string;
   };
+  [propName: string]: any; // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
I think I'd rather start with a `CreditCardType` which is more explicit and extend it to a `CustomCardType` which has an any property